pkg/reconciler/v1alpha1/autoscaling/kpa_scaler.go

```diff
@@ -127,6 +128,20 @@ func (ks *kpaScaler) Scale(ctx context.Context, kpa *kpa.PodAutoscaler, desiredS
 	}
 	currentScale := scl.Spec.Replicas
 
+	var serviceLabel string
+	var configLabel string
+	if kpa.Labels != nil {
+		serviceLabel = kpa.Labels[serving.ServiceLabelKey]
+		configLabel = kpa.Labels[serving.ConfigurationLabelKey]
+	}
+	reporter, err := autoscaler.NewStatsReporter(kpa.Namespace, serviceLabel, configLabel, kpa.Name)
+	if err != nil {
+		return err
+	}
+
+	reporter.Report(autoscaler.ActualPodCountM, float64(currentScale))
+	reporter.Report(autoscaler.RequestedPodCountM, float64(currentScale))
```

Comment:
@josephburnett the distinction here is unclear to me but since we're only looking at a `Scale` resource, it seems they don't have a `ReadyReplicas` value, which might be more appropriate for the `ActualPodCountM`?

Reply:
The most valuable thing that ReadyReplicas does is show clearly on the dashboard when cluster capacity is exhausted and where cluster autoscaling kicks in.  E.g. RequestedPodCount will be what we've asked of the scale resource.  But if its pods are unscheduled (due to cluster capacity shortfall) we want another metrics to show the number of pods actually in service.

But if you don't want to emit that from here, that's fine.  You could emit it from somewhere else that is aware of deployments (instead of just scaleref).

Reply:
The autoscaler has this information too, but not right here, but [here](https://github.com/knative/serving/blob/1e561bd96efa09d6cbc639c23c5de5871688dfd0/pkg/reconciler/v1alpha1/autoscaling/autoscaling.go#L208).

Reply:
If we want this too, perhaps we should move this code into `Reconcile()` and change the interface [here](https://github.com/knative/serving/blob/1e561bd96efa09d6cbc639c23c5de5871688dfd0/pkg/reconciler/v1alpha1/autoscaling/autoscaling.go#L190) to return the actual pod count?

Reply:
Done, moved the reporting code to appropriate places.